core/commands/cid.go

```diff
@@ -0,0 +1,307 @@
+package commands
+
+import (
+	"fmt"
+	"io"
+	"sort"
+	"strings"
+	"text/tabwriter"
+	"unicode"
+
+	cmds "gx/ipfs/QmPTfgFTo9PFr1PvPKyKoeMgBvYPh6cX3aDP7DHKVbnCbi/go-ipfs-cmds"
+	mhash "gx/ipfs/QmPnFwZ2JXKnXgMw8CdBPxn7FWh6LLdjUjxV1fKHuJnkr8/go-multihash"
+	cidutil "gx/ipfs/QmPyxJ2QS7L5FhGkNYkNcXHGjDhvGHueJ4auqAstFHYxy5/go-cidutil"
+	cmdkit "gx/ipfs/QmSP88ryZkHSRn1fnngAaV2Vcn63WUJzAavnRM9CVdU1Ky/go-ipfs-cmdkit"
+	mbase "gx/ipfs/QmSbvata2WqNkqGtZNg8MR3SKwnB8iQ7vTPJgWqB8bC5kR/go-multibase"
+	verifcid "gx/ipfs/QmVUhfewLZpSaAiBYCpw2krYMaiVmFuhr2iurQLuRoU6sD/go-verifcid"
+	cid "gx/ipfs/QmZFbDTY9jfSBms2MchvYM9oYRbAF19K7Pby47yDBfpPrb/go-cid"
+)
+
+var CidCmd = &cmds.Command{
+	Helptext: cmdkit.HelpText{
+		Tagline: "Convert and discover properties of CIDs",
+	},
+	Subcommands: map[string]*cmds.Command{
+		"format": cidFmtCmd,
+		"base32": base32Cmd,
+		"bases":  basesCmd,
+		"codecs": codecsCmd,
+		"hashes": hashesCmd,
+	},
+}
+
+var cidFmtCmd = &cmds.Command{
+	Helptext: cmdkit.HelpText{
+		Tagline: "Format and convert a CID in various useful ways.",
+		LongDescription: `
+Format and converts <cid>'s in various useful ways.
+
+The optional format string either "prefix" or or a printf style format string:
+` + cidutil.FormatRef,
+	},
+	Arguments: []cmdkit.Argument{
+		cmdkit.StringArg("cid", true, true, "Cids to format."),
+	},
+	Options: []cmdkit.Option{
+		cmdkit.StringOption("f", "Format string."),
+		cmdkit.StringOption("v", "CID version to convert to."),
+		cmdkit.StringOption("b", "Multibase to display CID in."),
+	},
+	Run: func(req *cmds.Request, resp cmds.ResponseEmitter, env cmds.Environment) {
+		fmtStr, _ := req.Options["f"].(string)
+		verStr, _ := req.Options["v"].(string)
+		baseStr, _ := req.Options["b"].(string)
+
+		opts := cidFormatOpts{}
+
+		switch fmtStr {
+		case "":
+			opts.fmtStr = "%s"
+		case "prefix":
+			opts.fmtStr = "%P"
```

Comment:
This feels a bit magical. What's the motivation? We could also just add a `ipfs cid prefix` command (alias of `ipfs cid format -f '%P' ...`).

Reply:
It was there in the original `cid-fmt` utility and copied over here.  For the `cid-fmt` utility is was the original use case (to get information about cids).  I am not tied to keeping it in this utility.

Reply:
I'd remove it for now. We can consider either a `prefix` or a `show` command later but we probably don't *need* them now.

Reply:
Will do.